Use WTF::map() in ScriptArguments::getArgumentsAsStrings() instead of uncheckedAppend()

cdumez opened the pull request:
#### 9152805b2bb446bffd04e3e661dc60fce7564cf1
<pre>
Use WTF::map() in ScriptArguments::getArgumentsAsStrings() instead of uncheckedAppend()
<a href="https://bugs.webkit.org/show_bug.cgi?id=262519">https://bugs.webkit.org/show_bug.cgi?id=262519</a>

Reviewed by Darin Adler.

Use WTF::map() in `ScriptArguments::getArgumentsAsStrings()` instead of `uncheckedAppend()`.
It is more efficient, now that `uncheckedAppend()` is just an alias to `append()`.

* Source/JavaScriptCore/inspector/ScriptArguments.cpp:
(Inspector::argumentAsString):
(Inspector::ScriptArguments::getArgumentAtIndexAsString const):
(Inspector::ScriptArguments::getArgumentsAsStrings const):

Canonical link: <a href="https://commits.webkit.org/268793@main">https://commits.webkit.org/268793@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6ee23948c80ffe8f4eb2b7793f70e45992b4b622

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20688 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/21106 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21762 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22583 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/19311 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24361 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21288 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20639 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20910 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20732 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/17987 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23439 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17897 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18793 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/25058 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/18018 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/18976 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/18971 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23004 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/20115 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19560 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/16619 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/24099 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18781 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/5742 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4965 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23118 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/25360 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19378 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5565 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/JavaScriptCore/inspector/ScriptArguments.cpp:
```diff
@@ -38,6 +38,18 @@
+    if (JSC::jsDynamicCast<JSC::ProxyObject*>(argument))
+        return "[object Proxy]"_s;
+
+    String result;
```
Does this have to be defined here instead of where it is initialized.

cdumez replied:
Oh, I just moved the code as is but I don't think there is a reason to defined it here. I'll update.

cdumez committed:
Use WTF::map() in ScriptArguments::getArgumentsAsStrings() instead of uncheckedAppend()
https://bugs.webkit.org/show_bug.cgi?id=262519

Reviewed by Darin Adler.

Use WTF::map() in `ScriptArguments::getArgumentsAsStrings()` instead of `uncheckedAppend()`.
It is more efficient, now that `uncheckedAppend()` is just an alias to `append()`.

* Source/JavaScriptCore/inspector/ScriptArguments.cpp:
(Inspector::argumentAsString):
(Inspector::ScriptArguments::getArgumentAtIndexAsString const):
(Inspector::ScriptArguments::getArgumentsAsStrings const):

Canonical link: https://commits.webkit.org/268793@main
